src/main/scala/akka/persistence/dynamodb/snapshot/DynamoDBSnapshotStore.scala

```diff
@@ -12,6 +12,8 @@ import com.typesafe.config.Config
 import akka.persistence.dynamodb._
 import scala.concurrent.Future
 
+class DynamoDBSnapshotRejection(message: String, cause: Throwable = null) extends RuntimeException(message, cause)
```

Comment:
Can we add a `Key` field as a `val` to this exception so we know which key caused the exception incase people want to catch it.

Also mention the key in the exception message.